src/swap/SwapAmountInput.tsx

```diff
@@ -66,143 +67,146 @@ const SwapAmountInput = ({
+              style={{ opacity: loading ? 0 : 1 }}
+              editable={editable && !loading}
+              keyboardType="decimal-pad"
+              // Work around for RN issue with Samsung keyboards
+              // https://github.com/facebook/react-native/issues/22005
+              autoCapitalize="words"
+              autoFocus={autoFocus}
+              // unset lineHeight to allow ellipsis on long inputs on iOS
+              inputStyle={[styles.inputText, inputError ? styles.inputError : {}]}
+              testID="SwapAmountInput/Input"
+              onBlur={() => {
+                handleSetStartPosition(0)
+              }}
+              onFocus={() => {
+                handleSetStartPosition(inputValue?.length ?? 0)
+              }}
+              onSelectionChange={() => {
+                handleSetStartPosition(undefined)
+              }}
+              selection={
+                Platform.OS === 'android' && typeof startPosition === 'number'
+                  ? { start: startPosition }
+                  : undefined
+              }
+            />
+            {loading && (
+              <View style={styles.loaderContainer}>
+                <SkeletonPlaceholder
+                  borderRadius={100} // ensure rounded corners with font scaling
+                  backgroundColor={Colors.gray2}
+                  highlightColor={Colors.white}
+                  testID="SwapAmountInput/Loader"
+                >
+                  <View style={styles.loader} />
+                </SkeletonPlaceholder>
+              </View>
+            )}
           </View>
-        )}
-      </View>
+          {!loading && parsedInputValue?.gt(0) && token && (
+            <Text style={styles.fiatValue} testID="SwapAmountInput/FiatValue">
+              <TokenDisplay
+                amount={parsedInputValue ?? 0}
+                showLocalAmount
+                showApprox
+                errorFallback={t('swapScreen.tokenUsdValueUnknown') ?? undefined}
+                tokenId={token?.tokenId}
+              />
+            </Text>
+          )}
+          {onPressMax && (
+            <View style={styles.maxButtonWrapper}>
+              <Touchable
+                borderRadius={Spacing.Tiny4}
+                onPress={() => {
+                  onPressMax()
+                  textInputRef.current?.blur()
+                }}
+                style={styles.maxButton}
+                testID="SwapAmountInput/MaxButton"
+              >
+                <Text style={styles.maxText}>{t('max')}</Text>
+              </Touchable>
+            </View>
+          )}
+        </View>
+      )}
     </View>
   )
 }
 
 const styles = StyleSheet.create({
   container: {
-    padding: Spacing.Regular16,
     backgroundColor: Colors.gray1,
     borderColor: Colors.gray2,
     borderWidth: 1,
   },
-  label: {
-    ...fontStyles.xsmall,
-    fontWeight: 'bold',
+  tokenInfo: {
+    flexDirection: 'row',
```

Comment:
i think you need an alignItems center here so that the icon is vertically centered. if you increase your device font size, the icon is no longer centered currently

Reply:
bump on this! :)